r/Dockerfile

```diff
@@ -0,0 +1,64 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+FROM arrow:cpp
+
+# get the stuff
+ADD r /arrow/r
+
+# Configure
+ENV CC=gcc \
+    CXX=g++
+
+# r-base includes tzdata. Get around interactive stop in that package
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Build R
+# [1] https://www.digitalocean.com/community/tutorials/how-to-install-r-on-ubuntu-18-04
+# [2] https://linuxize.com/post/how-to-install-r-on-ubuntu-18-04/#installing-r-packages-from-cran
+RUN apt install -y \
+    apt-transport-https \
+    software-properties-common && \
+    apt-key adv \
+        --keyserver keyserver.ubuntu.com \
+        --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9 && \
+    add-apt-repository 'deb https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/' && \
+    apt install -y r-base && \
+    # system libs needed by core R packages
+    apt install -y \
+        libgit2-dev \
+        libssl-dev && \
+    # Install vctrs from Github
+    Rscript -e "install.packages('devtools', repos = 'http://cran.rstudio.com')" && \
+    Rscript -e "devtools::install_github('r-lib/vctrs')" && \
+    # R is not good at picking up dependencies and installing them automatically
+    Rscript -e "install.packages(c( \
+        'Rcpp', \
+        'purrr', \
+        'assertthat', \
+        'fs', \
+        'tibble', \
+        'crayon', \
+        'testthat'), \
+        repos = 'http://cran.rstudio.com')"
+
+# build, install, test R package
+CMD arrow/ci/docker_build_cpp.sh && \
+    cd /arrow/r && \
+    R CMD build . && \
+    R CMD INSTALL $(ls | grep arrow_*.tar.gz) && \
+    R CMD check $(ls | grep arrow_*.tar.gz) --as-cran
```

Comment:
We mount arrow's source to `/arrow` and a `docker_cache` (host side) to `/build` (container side) and use `/build/cpp`, `/build/python`, `/build/c_glib` etc. to cache build files per language container. This enables us quicker edit build cycles.

Is it possible to define such a dir for R builds? See [cpp](https://github.com/apache/arrow/blob/master/ci/docker_build_cpp.sh#L30) for reference).

Reply:
This is really cool, thank you. Now I see what's going on!

Was able to get past a bug on install with this information (because it told me where to set `PKG_CONFIG_PATH` in the container)